src/cli/commands/force/lightning/lwc/preview.ts

```diff
@@ -142,146 +135,137 @@ export class Preview extends Setup {
             });
     }
 
-    protected async validateInputParameters(): Promise<void> {
-        return super.validateInputParameters().then(async () => {
-            const defaultDeviceName = CommandLineUtils.platformFlagIsIOS(
-                this.flags.platform
-            )
-                ? PlatformConfig.iOSConfig().defaultSimulatorName
-                : PlatformConfig.androidConfig().defaultEmulatorName;
-
-            this.deviceName = CommandLineUtils.resolveFlag(
-                this.flags.target,
-                defaultDeviceName
-            );
+    // TODO: Preview command takes quite a few command flags/parameters compared to other commands.
```

Comment:
Is there any reason not to just break the validate function out into its own standalone function, and reference it from the `validate` config? Or is the breakout more complex than that?

Reply:
Oh I'm forgetting that it would be a method for each option. Which will take a little more retooling than a simple one-shot extraction. I'm okay with us getting back to that in the future, personally.

Reply:
Since `flagsConfig` is a static property on the class I can only provide static method to `validate`. But static method won't have access to an instance object so anything that `validate` function validates can't assign the value to the object. What we can do, and I'm not sure if this is worth it, is to not use `CommandLineUtils.resolveFlags(...)` which checks the flag value's existence and length *AND* possibly assign a default value inside `validateInputParameter` but instead just check the value's existence and length in `validate`. If the `validate` didn't throw then we can assume that the value was there. As for default value we can have it initialized as such when the property is declared in the class. I didn't do this after consulting with Meisam because the code would follow similar pattern as others, however, there was nothing to be gained but potentially introducing a new bug.

Reply:
Yes, it's a bit complicated b/c of the way we compute `defaultDeviceName`. Setting the default device name is tricky b/c it  depends on platform.I also want to propose to remove that one altogether (in the future). We're using this default value b/c we are having the Preview command to either launch an emulator or create it in case it doesn't exists. But we now have a create command so we need to clean up Preview command to not create an emulator. The user can create one using the Create command if he needs to, then use the Preview command.